flang/include/flang/Common/idioms.h

```diff
@@ -87,6 +89,8 @@ template <typename... LAMBDAS> visitors(LAMBDAS... x) -> visitors<LAMBDAS...>;
 // To disable, compile with '-DCHECK=(void)'
 #ifndef CHECK
 #define CHECK(x) ((x) || (DIE("CHECK(" #x ") failed"), false))
+// Same as above, but with a custom error message.
+#define CHECK_MSG(x, y) ((x) || (DIE_MSG("CHECK(" #x ") failed: ", y), false))
```

Comment:
`#define CHECK_MSG(x, y) ((x) || (DIE("CHECK(" #x ") failed: " #y), false))` avoids a need for `DIE_MSG`.